### PR TITLE
Create conda environment

### PR DESCRIPTION
**Pull request by huguesv**
Add the most basic 'create conda environment' user interface to the environments window.

I concentrated on implementing and testing the various conda functionality we want to expose. I will now work on improving the UI, like show progress animation, show status after it's done, etc. and adding more functionality (delete, import, export, clone).

image

Add a "create conda environment" command to solution explorer that takes you directly to the ui controll with focus on name text box. Note that it doesn't add it to the project (at least right now), which is why I called the menu item "Create" instead of "Add". 

image

Both of the above are shown only when "detect conda environments" experimental option is on.

Show "Conda environment" as the description of the environment so that you can easily tell them apart in the list.

Fixes for conda 4.4 (which released this week) which no longer puts a conda.bat in created environments (we now look for a global one if the env doesn't have it).

When more than one root/base conda exists, we now check their version and use the latest one to enumerate and manage environments.

More functionality is implemented such as clone/export/import/delete, along with tests.

### CODE REVIEW

File: Python/Product/EnvironmentsList/Resources.resx [hunk 1]
```diff
@@ -123,6 +123,40 @@
...
+    <value>Create the environment.</value>
+  </data>
+  <data name="CondaExtensionEnvironmentNameLabel" xml:space="preserve">
+    <value>Name</value>
+  </data>
+  <data name="CondaExtensionEnvironmentNameHelp" xml:space="preserve">
+    <value>The name of the new environment.</value>
+  </data>
+  <data name="CondaExtensionEnvironmentNameWatermark" xml:space="preserve">
+    <value>e.g. My New Environment</value>
```
Comment (zooba) [inline Python/Product/EnvironmentsList/Resources.resx]: Should we offer "env" as the example? I think that's more common.

File: Python/Product/EnvironmentsList/images/images.xaml [hunk 2]
```diff
@@ -203,4 +203,15 @@
         </Setter>
         <Setter Property="VerticalAlignment" Value="Center" />
     </Style>
+
+    <Style x:Key="DeleteFolderImage" TargetType="Control">
+        <Setter Property="Template">
+            <Setter.Value>
+                <ControlTemplate>
+                    <imaging:CrispImage Moniker="{x:Static catalog:KnownMonikers.DeleteFolder}" />
```
Comment (zooba) [inline Python/Product/EnvironmentsList/images/images.xaml]: Should we get a "delete environment" icon created? I kind of like the implication that an actual folder will be deleted though.

File: Python/Product/PythonTools/PythonTools/InterpreterList/InterpreterListToolWindow.cs [hunk 3]
```diff
@@ -394,6 +410,60 @@ sealed class InterpreterListToolWindow : ToolWindowPane {
...
+                MessageBoxImage.Question
+            );
+            if (result != MessageBoxResult.Yes) {
+                return;
+            }
+
+            var compModel = _site.GetService(typeof(SComponentModel)) as IComponentModel;
+            var registry = compModel.GetService<IInterpreterRegistryService>();
+            // TODO: reuse a single conda environment manager between this and
+            // the one in CondaExtension, since it can be slow to create
```
Comment (zooba) [inline Python/Product/PythonTools/PythonTools/InterpreterList/InterpreterListToolWindow.cs]: Better off making it fast to create, or just make it a MEF export and then get it through the component model.

File: Python/Product/PythonTools/PythonTools/Project/InterpretersContainerNode.cs [hunk 4]
```diff
@@ -24,6 +24,7 @@
 using VsMenus = Microsoft.VisualStudioTools.Project.VsMenus;
 using Microsoft.VisualStudio.Imaging;
 using Microsoft.VisualStudio.Imaging.Interop;
+using Microsoft.PythonTools.Interpreter;
```
Comment (zooba) [inline Python/Product/PythonTools/PythonTools/Project/InterpretersContainerNode.cs]: Sort

File: Python/Product/PythonTools/PythonTools/Project/PythonProjectNode.cs [hunk 5]
```diff
@@ -1652,6 +1652,9 @@ private static T FindImmediateChild<T>(HierarchyNode parent)
                     case PythonConstants.AddEnvironment:
                         ShowAddInterpreter();
                         return VSConstants.S_OK;
+                    case PythonConstants.CreateCondaEnv:
+                        ExecCreateCondaEnv();
```
Comment (zooba) [inline Python/Product/PythonTools/PythonTools/Project/PythonProjectNode.cs]: `return ExecCreateCondaEnv`

File: Python/Product/VSInterpreters/Interpreter/CondaEnvironmentFactoryProvider.cs [hunk 6]
```diff
@@ -337,7 +327,7 @@ class CondaInfoResult {
             return unique;
         }
 
-        #region IPythonInterpreterProvider Members
+#region IPythonInterpreterProvider Members
```
Comment (zooba) [inline Python/Product/VSInterpreters/Interpreter/CondaEnvironmentFactoryProvider.cs]: Nit: revert

File: Python/Product/VSInterpreters/PackageManager/CPythonCondaPackageManagerProvider.cs [hunk 7]
```diff
@@ -17,16 +17,43 @@
 using System;
 using System.Collections.Generic;
 using System.ComponentModel.Composition;
-using Microsoft.PythonTools.Infrastructure;
+using System.IO;
```
Comment (zooba) [inline Python/Product/VSInterpreters/PackageManager/CPythonCondaPackageManagerProvider.cs]: Sort

Comment (huguesv) [reply Python/Product/VSInterpreters/PackageManager/CPythonCondaPackageManagerProvider.cs]: already sorted, that Microsoft line was deleted

File: Python/Product/EnvironmentsList/images/images.xaml (hunk 2 above)
Comment (huguesv) [reply Python/Product/EnvironmentsList/images/images.xaml]: I figured the folder one was pretty good, where as a custom delete environment may still be unclear that it deletes it and not just removes it from the list.

File: Python/Product/EnvironmentsList/Resources.resx (hunk 1 above)
Comment (huguesv) [reply Python/Product/EnvironmentsList/Resources.resx]: sure

File: Python/Product/PythonTools/PythonTools/InterpreterList/InterpreterListToolWindow.cs (hunk 3 above)
Comment (huguesv) [reply Python/Product/PythonTools/PythonTools/InterpreterList/InterpreterListToolWindow.cs]: Yeah, MEF with some caching of previously seen conda environments along with their `conda -V` version will be helpful. I'll file a new issue for it.

**Pull request merged**